### PR TITLE
change client name

### PR DESCRIPTION
**Pull request by andrewyuq**
Screenshot 2024-04-25 at 1 16 49 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt [hunk 1]
```diff
@@ -116,14 +118,17 @@ class SsoAccessTokenProvider(
     @Deprecated("Device authorization grant flow is deprecated")
     private fun registerDAGClient(): ClientRegistration {
         loadDagClientRegistration()?.let {
+            // TODO: remove this logic after TARGET_SAFE_CLIENT_NAME_DATE
+            // client registration was created before 4/30
+            if (it.expiresAt.isBefore(TARGET_SAFE_CLIENT_NAME_DATE)) return@let
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:123]: let's leave this part out

Comment (andrewyuq) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt:123]: ok, I will revert it.

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt [hunk 2]
```diff
@@ -3,13 +3,25 @@
 
 package software.aws.toolkits.jetbrains.services.amazonq.startup
 
+import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.startup.ProjectActivity
+import com.intellij.openapi.wm.ToolWindowManager
+import migration.software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhispererExplorerActionManager
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt:10]:
```suggestion
import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhispererExplorerActionManager
```

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt [hunk 3]
```diff
@@ -3,13 +3,25 @@
...
 import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AmazonQToolWindow
+import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AmazonQToolWindowFactory
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhispererExplorerActionManager
 
 class AmazonQStartupActivity : ProjectActivity {
     override suspend fun execute(project: Project) {
         // initialize html contents in BGT so users don't have to wait when they open the tool window
         AmazonQToolWindow.getInstance(project)
+
+        if (CodeWhispererExplorerActionManager.getInstance().getIsFirstRestartAfterQInstall()) {
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt:19]: for 241 failure, maybe move the application service declaration to `module-amazonq.xml` from `plugin-codewhisperer.xml`

Comment (andrewyuq) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt:19]: it works

**Pull request merged**